.gitlab-ci.yml

```diff
@@ -39,18 +206,120 @@ nix-dry:
     - nix-build -v -v --dry-run ./release.nix --attr package.macos.x64.macho
 
 nix:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
   stage: build
   interruptible: true
   script:
     - >
-        nix-build ./release.nix
-        --max-jobs $(nproc)
-        --attr application
-        --attr docker
+      BUILDOUT=`nix-build ./release.nix\
+        --attr application\
+        --attr docker\
+        --attr package.linux.x64.elf\
+        --attr package.windows.x64.exe\
+        --attr package.macos.x64.macho`;
+        mkdir builds;
+        cp -r $BUILDOUT builds/;
+        chmod -R a+rw builds;
   only:
     - master
+  artifacts:
+    paths:
+      - builds
+
+runs in node:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
+  stage: quality
+  only:
+    - master
+  interruptible: true
+  script:
+    - >
+      nix-shell -I nixpkgs=./pkgs.nix --packages nodejs --run '
+      npm install;
+      echo password > testPassword;
+      npm run polykey -- agent start --test -np ./nodePath --password-file ./testPassword;
+      '
+
+running application build:
+  image: registry.gitlab.com/matrixai/engineering/maintenance/gitlab-runner
+  stage: quality
+  only:
+    - master
+  needs:
+    - job: nix
+      artifacts: true
+  interruptible: true
+  script:
+    - >
+      nix-shell -I nixpkgs=./pkgs.nix --packages nodejs --run '
+      echo password > testPassword;
+      node builds/*/lib/node_modules/@matrixai/polykey/dist/bin/polykey.js agent start -np ./nodePath --password-file ./testPassword --test;
+      '
+
+
+running docker build:
+  image: docker:19.03.12
+  variables:
+    DOCKER_TLS_CERTDIR: "/certs"
+  services:
+    - docker:19.03.12-dind
+  before_script:
+    - docker info
+  stage: quality
+  only:
+    - master
+  needs:
+    - job: nix
+      artifacts: true
+  interruptible: true
+  script:
+    - ./scripts/runDocker.sh
+
+running linux build:
+  stage: quality
+  only:
+    - master
+  tags:
+    - linux
+  needs:
+    - job: nix
+      artifacts: true
+  interruptible: true
+  script:
+    - "echo password > testPassword"
+    - "./builds/*linux* agent start -np ./nodePath --password-file ./testPassword --test"
+
+running windows build:
+  stage: quality
+  only:
+    - master
+  tags:
+    - windows
+  needs:
+    - job: nix
+      artifacts: true
+  interruptible: true
+  script:
+    - "echo password > testPassword"
+    - "./builds/*win32* agent start -np ./nodePath --password-file ./testPassword --test"
+
+running macos build:
+  tags: [ shared-macos-amd64 ]
```

Comment:
Make this consistent with the other `tags` specifiers.

Reply:
Fixed.